packages/gitbook/src/components/SitePage/SitePage.tsx

```diff
@@ -133,6 +136,25 @@ export async function generateSitePageMetadata(props: SitePageProps): Promise<Me
     const { site, customization, revision, linker, imageResizer } = context;
     const siteStructureTitle = getSiteStructureTitle(context);
 
+    const canonical = pageMetaLinks?.canonical
+        ? pageMetaLinks.canonical
+        : // Trim trailing slashes in canonical URL to match the redirect behavior
+          linker
+              .toAbsoluteURL(linker.toPathForPage({ pages: revision.pages, page }))
+              .replace(/\/+$/, '');
```

Comment:
I can't remember why we have this; maybe it should be done on the other canonical as well